# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.h
# Commit: d55b6525d7b2
# Full Hash: d55b6525d7b24d90ac7e1b092e4fef698cc244fe
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-13 04:28:43
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r 26f637a20dc1 -r d55b6525d7b2 dom/cookiestore/CookieChangeEvent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/cookiestore/CookieChangeEvent.h	Thu Sep 12 16:48:18 2024 +0000
@@ -0,0 +1,41 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_CookieChangeEvent_h
+#define mozilla_dom_CookieChangeEvent_h
+
+#include "mozilla/dom/Event.h"
+#include "mozilla/dom/BindingUtils.h"
+#include "mozilla/dom/CookieChangeEventBinding.h"
+
+namespace mozilla::dom {
+
+class CookieChangeEvent final : public Event {
+ public:
+  CookieChangeEvent(EventTarget* aOwner, nsPresContext* aPresContext,
+                    WidgetEvent* aEvent);
+
+  JSObject* WrapObjectInternal(JSContext* aCx,
+                               JS::Handle<JSObject*> aGivenProto) override;
+
+  static already_AddRefed<CookieChangeEvent> Constructor(
+      const GlobalObject& aGlobal, const nsAString& aType,
+      const CookieChangeEventInit& aEventInit);
+
+  void GetChanged(nsTArray<CookieListItem>& aList) const;
+
+  void GetDeleted(nsTArray<CookieListItem>& aList) const;
+
+ private:
+  ~CookieChangeEvent();
+
+  nsTArray<CookieListItem> mChanged;
+  nsTArray<CookieListItem> mDeleted;
+};
+
+}  // namespace mozilla::dom
+
+#endif /* mozilla_dom_CookieChangeEvent_h */